Removed a few defunct Gradle Settings

lkishalmi opened the pull request:
Removed the following no-more useful Settings from Gradle:

- Start Gradle Daemon on IDE Start
  In the early days (2.x,3.x) it could have saved a few seconds, by loading the Daemon early, however in those days the Wrapper was not commonly used. Usually every project were used with one Gradle version set in the IDE. Project loading was also done by that daemon. Today, there is no advantage to fire up a Gradle Daemon which most likely won't be used.
- Install Gradle Runtime Silently
  It started to download a new Gradle version when it was selected in the settings dialog right when the dialog settings were applied. It could save a few seconds in the old days. Today it is usually done by the wrapper, and the project trust replaced this functionality.
- Do not Rebuild Project Dependencies
  In the older days, this was a real performance boost working on projects. Right now the flag state is, I'd say undecided. It got removed in `Gradle 8.0` without deprecation, got restored in `Gradle 8.0.1`. There is no sigh of deprecation since then. We put it in deprecated on NB18 for the option removal in NB20. It still can be used if someone overwrite the Gradle actions, but it's better to leave this option out.

![image](https://github.com/apache/netbeans/assets/1381701/52ee2f26-7558-464d-90df-e550a5505f68)

mbien reviewed (approved):
good cleanup. I quickly checked the layout and everything resized fine.

sdedic commented on extide/gradle/apichanges.xml:
```diff
@@ -83,6 +83,21 @@ is the proper place.
+            <version major="2" minor="36"/>
+            <date day="2" month="10" year="2023"/>
+            <author login="lkishalmi"/>
+            <compatibility semantic="compatible" />
```
IMHO semantic should be `incompatible`: code that hopes the setter (setting) has some effect may break.

sdedic commented on extide/gradle/src/org/netbeans/modules/gradle/spi/GradleSettings.java:
```diff
@@ -182,12 +185,24 @@ public File getGradleUserHome() {
+     * @param b
+     * @deprecated
+     */
+    @Deprecated
```
Consider to log a `WARNING`-level message in the setters.

mbien commented:
wanted to merge but seeing two unresolved reviewer comments, freeze is on the 17th.

lkishalmi commented:
I've started to work on removing the Grade version selector and adding a default java runtime instead. I hope I can submit that work on this branch tomorrow.

lkishalmi commented:
![image](https://github.com/apache/netbeans/assets/1381701/e81ab645-e7d9-4343-9d70-81b35230ff5d)

Further cleanup, removed the option to select the Gradle version, added default Java Runtime selection for Gradle execution.

There could be further enhancements, around warnings and auto selecting the proper JDK for Gradle Execution, but this is how far I could go before NB20 window closes.

mbien commented:
as discussed on apache slack, going to add a commit which fixes some minor layout issues. Some show up right away, some are only visible when the window is resized:
![layout-issues-gradle1](https://github.com/apache/netbeans/assets/114367/0d3033e2-0c3d-43ff-bd21-edc1daaf3a9a)
![layout-issues-gradle2](https://github.com/apache/netbeans/assets/114367/d80a7309-fc33-4d75-8df0-232f5f96d071)
![layout-issues-gradle3](https://github.com/apache/netbeans/assets/114367/6300d35a-db69-4455-aeee-09e3aaf340ca)
